engine/access/rest/rest_api.go

```diff
@@ -0,0 +1,88 @@
+package rest
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RestAPIHandler provides the implementation of each of the REST API
+type RestAPIHandler struct {
+	backend *backend.Backend
+	logger  zerolog.Logger
+	encoder encoding.Encoder
+}
+
+func NewRestAPIHandler(backend *backend.Backend, logger zerolog.Logger) *RestAPIHandler {
+	return &RestAPIHandler{
+		backend: backend,
+		logger:  logger,
+		encoder: encoding.DefaultEncoder, //use the default JSON encoder
```

Comment:
Are we sure this is needed? I would have assumed that the generated server code would have provided some way of automatically serializing the generated structs (e.g `generated.Block`)?

Reply:
actually the generated code did not import the json encoder. They don't have a fully working [Go server example](https://github.com/swagger-api/swagger-codegen/blob/master/samples/server/petstore/go-api-server/go/api_pet.go) but the [client code example](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/go/go-petstore/client.go#L16) that they have does import `encoding/json`